mesh/mesh_connections.hpp

```diff
@@ -0,0 +1,256 @@
+      |   |   |      9  10  11      | 2 | 3 |      6   |   7
+      7---8---9      +-4-+-5-+      +---+---+      +-2-+-3-+ 
+
+                        child        parents
+                      dim,bdry,id  dim,bdry,ids
+      ParentsOfEntity({1,false,7}, {2,false,{}}) ->  {2,false,{0,1}}
+      ParentsOfEntity({1,true ,5}, {2,false,{}}) ->  {2,false,{1}}**/
+   void ParentsOfEntity(const EntityID &child, EntityIDs &parents);
+
+   /** @brief Returns the parent entity IDs of any of the given @a children
+       entities.  The @a parents struct operates as an in/out 
+       parameter.  The kind of the parent entities is set in the 
+       @a parents.dim and @a parents.boundary members.  The parent 
+       entity ID numbers are placed in @a parents.ids member.  For the 
+       example mesh with vertex, edge, face/element, and boundary element 
+       IDs we have:
+
+       dim = 0        dim = 1        dim = 2     Boundary Elems 
+      0---1---2      +-0-+-1-+      +---+---+      +-0-+-1-+
+      |   |   |      6   7   8      | 0 | 1 |      4   |   5
+      3---4---5      +-2-+-3-+      +---+---+      +---+---+
+      |   |   |      9  10  11      | 2 | 3 |      6   |   7
+      7---8---9      +-4-+-5-+      +---+---+      +-2-+-3-+ 
+
+                             children          parents
+                           dim,bdry,ids     dim,bdry,ids
+      ParentsOfAnyEntities({0,false,{0,9}}, {2,false,{}}) ->  {2,false,{0,3}}
+      ParentsOfAnyEntities({1,true ,{4,6}}, {2,false,{}}) ->  {2,false,{0,2}}**/
+   void ParentsOfAnyEntities(const EntityIDs &children, EntityIDs &parents);
+
+   /** @brief Returns the parent entity IDs that have all their child entities
+       covered by the @a children entities.  The @a parents struct operates as 
+       an in/out parameter.  The kind of the parent entities is set in the 
+       @a parents.dim and @a parents.boundary members.  The parent 
+       entity ID numbers are placed in @a parents.ids member.  For the 
+       example mesh with vertex, edge, face/element, and boundary element 
+       IDs we have:
+
+       dim = 0        dim = 1        dim = 2     Boundary Elems 
+      0---1---2      +-0-+-1-+      +---+---+      +-0-+-1-+
+      |   |   |      6   7   8      | 0 | 1 |      4   |   5
+      3---4---5      +-2-+-3-+      +---+---+      +---+---+
+      |   |   |      9  10  11      | 2 | 3 |      6   |   7
+      7---8---9      +-4-+-5-+      +---+---+      +-2-+-3-+ 
+
+                                  children              parents
+                                dim,bdry,ids          dim,bdry,ids
+      ParentsCoveredByEntities({1,false,{0,1,2,6,7}}, {2,false,{}}) ->  {2,false,{0}}, element 1 is not covered
+      ParentsCoveredByEntities({0,false,{0,1,2,3,4}}, {2,false,{}}) ->  {2,false,{0}}
+      ParentsCoveredByEntities({0,false,{0,1,2,3,7}}, {1,true ,{}}) ->  {1,true ,{0,1,4,6}}**/   
+   void ParentsCoveredByEntities(const EntityIDs &children, EntityIDs &parents);
```

Comment:
I'm not sure I understand the example:
```
ParentsCoveredByEntities({1,false,{0,1,2,6,7}}, {2,false,{}}) -> {2,false,{0}}, element 1 is not covered
```
Why is `0` part of the result? Since `parents.boundary == false`, I would expect the result to be `{}`, since `0` is a boundary element.